src/request.ts

```diff
@@ -891,13 +891,26 @@ class DatastoreRequest {
     query: Query,
     options: RunQueryStreamOptions = {}
   ): SharedQueryOptions {
+    // TODO: This needs to be refactored later:
+    // TODO: Create a separate function (getSharedOptionsOnly) that only accepts options as a parameter and do two the first two if statements in it
+    // TODO: Call getSharedOptionsOnly here and in makeRequest in createReadStream
     const sharedQueryOpts = {} as SharedQueryOptions;
     if (options.consistency) {
       const code = CONSISTENCY_PROTO_CODE[options.consistency.toLowerCase()];
       sharedQueryOpts.readOptions = {
         readConsistency: code,
       };
     }
+    if (options.readTime) {
+      if (sharedQueryOpts.readOptions === undefined) {
+        sharedQueryOpts.readOptions = {};
+      }
+      const readTime = options.readTime;
+      const seconds = readTime / 1000;
+      sharedQueryOpts.readOptions.readTime = {
+        seconds: Math.floor(seconds),
+      };
+    }
```

Comment:
This code was taken from https://github.com/googleapis/nodejs-datastore/blob/e7c3aaf665ebae5a91fb692bb0af54717e5ea587/src/request.ts#L290-L299 and can be refactored after the merge as per the TODO.

Reply:
any reason why we can't refactor this now?

Reply:
No reason. It is a choice between having two simpler PRs and one PR that includes both changes. Doing so now is fine though.